src/ParallelTempering.jl

```diff
@@ -126,13 +180,121 @@ function gather_print(replica::Replica, data, fname::String)
     return nothing
 end
 
+"""
+Print xyz formatted (Lx, Ly, Lz, Sx, Sy, Sz) configurations to file 
+"""
+function xyz_to_file(sys::SpinSystem{3}, output::IOStream)
+    sites = reinterpret(reshape, Float64, collect(sys.lattice))
+    spins = reinterpret(reshape, Float64, collect(sys.sites))
+
+    xyz = hcat(
+        hcat([vec(sites[i,:,:,:,:]) for i in 1:3]...),
+        hcat([vec(spins[i,:,:,:,:]) for i in 1:3]...)
+    )
+
+    for r in eachrow(xyz)
+        @printf(output, "%f\t%f\t%f\t%f\t%f\t%f\n", r...)
+    end
+    println(output, "")
+
+    return nothing
+end
+
+""" 
+Update temperature schedule using feedback-optimized parallel tempering from
+Katzgraber et. al. (https://arxiv.org/pdf/cond-mat/0602085v3.pdf).
 
+Note: use f(T) that goes from T_max -> T_min instead of original paper
+Note: use (k-1)/(M-1) as target for integral in Eq. 11 instead of k/M 
 """
-Start a parallel tempering (PT) simulation. 
-Run w/ the command "mpiexec -n [n_procs] julia --project [julia_script.jl]".
+function feedback_update!(replica::Replica, N_updates::Int64; w::Float64=0.0, dT′::Float64=1e-4)
+
+    T_new = zeros(replica.N_ranks)
+
+    N_labeled = (replica.N_up + replica.N_down)    
+    f = ((N_labeled > 0) ? replica.N_down/N_labeled : 0.0)
+    
+    # gather temperatures and replica flow 'f' from all replicas
+    gv = MPI.Gather([1.0/replica.sampler.β, f], 0, MPI.COMM_WORLD)
```

Comment:
Tiny nitpick, but maybe a better variable name than `gv`?

Reply:
I'm also realizing that `replica.sampler.β` will error with Langevin sampling too, and I've provided no way to get the temperature back out of a `LangevinSampler`! I can add another commit to this PR adding a new function like `get_temp` for all samplers that returns their temperature that we can call here instead -- I'll change this and other lines that grab the temperature in my commit.

So that we don't get conflicts, I'll wait until after you've made your pass?

Reply:
sounds good